cmake/configure/configure_1_threads.cmake

```diff
@@ -138,6 +138,24 @@ MACRO(FEATURE_THREADS_FIND_EXTERNAL var)
   IF(TBB_FOUND)
     SET(${var} TRUE)
   ENDIF()
+
+  #
+  # TBB versions before 4.2 are missing some explicit calls to std::atomic::load
+  # in ternary expressions; these cause compilation errors in some compilers
+  # (such as GCC 8.1 and newer). To fix this we simply blacklist all older
+  # versions:
+  #
+  IF(TBB_VERSION_MAJOR LESS 4 OR
+      ((TBB_VERSION_MAJOR EQUAL 4) AND (TBB_VERSION_MINOR LESS 2)))
+    MESSAGE(STATUS "Could not find a sufficiently modern TBB installation: "
+      "Version >= 4.2.0 required!"
```

Comment:
In other cases, we don't error out but simply and silently fall back to the bundled version here. Why not do that here?

Reply:
Good point; we should do that. I copied the same logic that we use in boost that should fall back to the internal version but I don't have an easy way to check that this works right now.